kiwi/boot/image/dracut.py

```diff
@@ -98,6 +116,7 @@ def write_system_config_file(self, config, config_file=None):
             config_file = os.path.normpath(
                 self.boot_root_directory + Defaults.get_dracut_conf_name()
             )
+        assert config_file
```

Comment:
why the assert ?

Actually in any of the changes here the behavior of the code should stay untouched

Reply:
This is a mypy issue: it isn't able to infer that `config_file` is never `None`. The `assert` is essentially only needed as a type hint, which should be ok, because that will be removed when byte compiling this file anyway.

Reply:
If we agree to base the type hints on py >= 3.6 as discussed a bit on the rio chat, would this still apply ? I think we should not add code to make mypy happy, thoughts ?